homeassistant/components/device_tracker/tplink.py

```diff
@@ -52,9 +52,12 @@ def get_scanner(hass, config):
             TplinkDeviceScanner, Tplink5DeviceScanner, Tplink4DeviceScanner,
             Tplink3DeviceScanner, Tplink2DeviceScanner, Tplink1DeviceScanner
     ]:
-        scanner = cls(config[DOMAIN])
-        if scanner.success_init:
-            return scanner
+        try:
+            scanner = cls(config[DOMAIN])
+            if scanner.success_init:
+                return scanner
+        except ConnectionError:
+            _LOGGER.warning("ConnectionError in scanner %s", cls.__name__)
```

Comment:
`_LOGGER.debug()` is enough, this exception was expecting since some device only be able to init as specific type of scanner.

Reply:
And then remove this, but use `_LOGGER.debug()` in these individual except blocks?